step-grid-api/src/main/java/step/grid/agent/tokenpool/TokenReservationSession.java

```diff
@@ -42,4 +47,26 @@ public List<TokenEventListener> getEventListeners() {
     public void clearEventListeners() {
         eventListeners.clear();
     }
+
+    /**
+     * This method is allowed also for fake sessions using the UnusableTokenReservationSession class.
+     * It is meant to store technical objects (application context, file version) following the lifecycle of sessions.
+     * When keywords are executed outside a session context, the UnusableTokenReservationSession is used and is closed from the AgentTokenPool.afterTokenExecution hook
+     * @param closeable to be stored
+     */
+    public <T extends AutoCloseable> void closeWithSession(T closeable) {

```

Comment:
Very minor but the method name is a bit misleading as it suggests an immediate action "close". Replacing by an explicit method name "trackObjectToBeClosedWithSession" or similar could make it easier to understand..